### PR TITLE
Block editor

### LINKED ISSUE #20678: Create storyblock field

### PR DESCRIPTION
**Pull request by erickgonzalez**
Create a new field called `StoryBlock`.
Since what will be stored is a JSON, we need a StoryBlockMap to get each of the 3 properties (render, content, type).
Logic to get the field from GraphQL, StoryBlock must behave as an object to get 2 properties (render and JSON, JSON will not include the render).

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/contenttype/model/field/StoryBlockField.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.google.common.collect.ImmutableList;
+import java.util.Collection;
+import java.util.List;
+import org.immutables.value.Value;
+
+@JsonSerialize(as = ImmutableStoryBlockField.class)
+@JsonDeserialize(as = ImmutableStoryBlockField.class)
+@Value.Immutable
+public abstract class StoryBlockField extends Field {
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/contenttype/model/field/StoryBlockField.java:16]: Add doc explaining it

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/content/StoryBlockMap.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import com.dotcms.contenttype.transform.field.LegacyFieldTransformer;
+import com.dotcms.repackage.org.codehaus.jettison.json.JSONException;
+import com.dotcms.repackage.org.codehaus.jettison.json.JSONObject;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.portlets.structure.model.Field;
+import com.liferay.util.StringPool;
+import org.apache.commons.lang.builder.ToStringBuilder;
+
+public class StoryBlockMap {
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/content/StoryBlockMap.java:13]: Doc it

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/content/StoryBlockMap.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+    private String type = StringPool.BLANK;
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/content/StoryBlockMap.java:15]: it seems it could be final seems do not have any set

**Review by jdotcms — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/content/StoryBlockMap.java (hunk 3 above)
Comment (erickgonzalez) [reply dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/content/StoryBlockMap.java:15]: it's set over here https://github.com/dotCMS/core/pull/20785/files#diff-7aa369151c4492eaa1284343818cee85b0f7b92966ac27e6a14a58a4da69e0e0R23

**Review by dsilvam — APPROVED** [review]

**Review by jdotcms — APPROVED** [review]

**Review by jdotcms — APPROVED** [review]

**Pull request merged**